app/views/posts/index.html.erb

```diff
@@ -12,14 +12,29 @@
   <tbody>
     <% @posts.each do |post| %>
       <tr>
+      <td><h1><%= post.title %></h1>
+
+      <p><%= post.body %></p>
+      <% if user_signed_in?%>
+      <p>Written by <%= post.author%></p>
+      <% end %></td>
```

Comment:
**Optional**

- [ ] It will be nice to move this logic to a helper. There should be absolutely NO logic in views. 
Views are only for displaying information that became available to it from its corresponding controller not generating it. Check this [article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) if you need help on how to use a helper in rails.